packages/cli/src/services/ai/providers/openai.ts

```diff
@@ -0,0 +1,35 @@
+import config from '@/config';
+import { ChatOpenAI } from '@langchain/openai';
+import type { BaseMessageChunk, BaseMessageLike } from '@langchain/core/messages';
+import type { N8nAIProvider } from '@/types/ai.types';
+
+export class AIProviderOpenAI implements N8nAIProvider {
+	private model: ChatOpenAI;
+
+	constructor() {
+		this.model = new ChatOpenAI({
+			openAIApiKey: config.getEnv('ai.openAIApiKey'),
+			modelName: 'gpt-3.5-turbo-16k',
+			timeout: 60000,
+			maxRetries: 2,
+		});
+	}
+
+	mapResponse(data: BaseMessageChunk): string {
+		if (Array.isArray(data.content)) {
+			return data.content
+				.map((message) =>
+					'text' in message ? message.text : 'image_url' in message ? message.image_url : '',
```

Comment:
Suggested fix didn't work since typescript needs to be able to determine the type from 3 different response types. I'm not sure when the other two (array case) happen but I thought I'd handle them anyway.